pkg/os/os-mapper-provider_test.go

```diff
@@ -0,0 +1,164 @@
+package os_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	osmap "github.com/kubevirt/vm-import-operator/pkg/os"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	get func(context.Context, client.ObjectKey, runtime.Object) error
+)
+
+var _ = Describe("OS Map Provider ", func() {
+	var (
+		mClient  *mockClient
+		osMapper *osmap.OSMaps
+	)
+
+	BeforeEach(func() {
+		mClient = &mockClient{}
+		osMapper = osmap.NewOSMapProvider(mClient)
+	})
+
+	Describe("Core OS Map only", func() {
+		It("should load OS map", func() {
+			guestOsToCommon, osInfoToCommon, err := osMapper.GetOSMaps()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(guestOsToCommon).NotTo(BeEmpty())
+			Expect(osInfoToCommon).NotTo(BeEmpty())
+		})
+	})
+
+	Describe("User OS Map env vars set with empty value", func() {
+		BeforeEach(func() {
+			os.Setenv(osmap.OsConfigMapName, "")
+			os.Setenv(osmap.OsConfigMapNamespace, "")
+		})
+
+		It("should load OS map", func() {
+			guestOsToCommon, osInfoToCommon, err := osMapper.GetOSMaps()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(guestOsToCommon).NotTo(BeEmpty())
+			Expect(osInfoToCommon).NotTo(BeEmpty())
+		})
+	})
+
+	Describe("User OS Map env vars set with non-empty value", func() {
+		BeforeEach(func() {
+			os.Setenv(osmap.OsConfigMapName, "osmap-name")
+			os.Setenv(osmap.OsConfigMapNamespace, "osmap-namespace")
+		})
+
+		It("should fail when user OS map does not exist", func() {
+			get = func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+				return fmt.Errorf("Not found")
+			}
+
+			guestOsToCommon, osInfoToCommon, err := osMapper.GetOSMaps()
+
+			Expect(err).To(HaveOccurred())
+			// core values should be loaded prior to loading user OS map data
+			Expect(guestOsToCommon).NotTo(BeEmpty())
+			Expect(osInfoToCommon).NotTo(BeEmpty())
+		})
+
+		It("should prefer user OS map values", func() {
+			get = func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+				obj.(*corev1.ConfigMap).Data = make(map[string]string)
+				obj.(*corev1.ConfigMap).Data[osmap.GuestOsToCommonKey] = `"Fedora": "myfedora"`
+				obj.(*corev1.ConfigMap).Data[osmap.OsInfoToCommonKey] = `"rhel_6_9_plus_ppc64": "myos"`
+				return nil
+			}
+
+			guestOsToCommon, osInfoToCommon, err := osMapper.GetOSMaps()
+
+			Expect(err).NotTo(HaveOccurred())
+			// core values should be loaded prior to loading user OS map data
```

Comment:
Depending of the intended behaviour of I would suggest following to make the API behaviour explicitly described:
- if the map contents are merged - add check that one of the core mappings still works;
- otherwise check that one of the core ones is not there;